pkg/server/server.go

```diff
@@ -48,31 +49,15 @@ func GetInstance(serverConfig *ServerInstanceConfig) *Server {
 			log.Fatal("Server not initialized")
 		}
 		server = &Server{
-			config:                   serverConfig.Config,
-			TailscaleServer:          serverConfig.TailscaleServer,
-			ProviderTargetService:    serverConfig.ProviderTargetService,
-			ContainerRegistryService: serverConfig.ContainerRegistryService,
-			WorkspaceService:         serverConfig.WorkspaceService,
-			ApiKeyService:            serverConfig.ApiKeyService,
-			GitProviderService:       serverConfig.GitProviderService,
-			ProviderManager:          serverConfig.ProviderManager,
-			ProfileDataService:       serverConfig.ProfileDataService,
+			ServerInstanceConfig: *serverConfig,
 		}
 	}
 
 	return server
 }
 
 type Server struct {
-	config                   Config
-	TailscaleServer          TailscaleServer
-	ContainerRegistryService containerregistries.IContainerRegistryService
-	ProviderTargetService    providertargets.IProviderTargetService
-	WorkspaceService         workspaces.IWorkspaceService
-	ApiKeyService            apikeys.IApiKeyService
-	GitProviderService       gitproviders.IGitProviderService
-	ProviderManager          manager.IProviderManager
-	ProfileDataService       profiledata.IProfileDataService
+	ServerInstanceConfig
```

Comment:
We talked about how `server.Config` shouldn't be exposed publicly and that's why we had
```golang
config                   Config
TailscaleServer          TailscaleServer
ContainerRegistryService containerregistries.IContainerRegistryService
ProviderTargetService    providertargets.IProviderTargetService
WorkspaceService         workspaces.IWorkspaceService
ApiKeyService            apikeys.IApiKeyService
GitProviderService       gitproviders.IGitProviderService
ProviderManager          manager.IProviderManager
ProfileDataService       profiledata.IProfileDataService
```